lib/hexpm/accounts/pwned/hexpm.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Hexpm.Accounts.Pwned.Hexpm do
+  @behaviour Hexpm.Accounts.Pwned
+
+  @base_url "https://api.pwnedpasswords.com/"
+  @weakness_threshold 1
+  @timeout 500
+
+  @spec password_breached?(String.t()) :: boolean
+  def password_breached?(string_password) do
+    string_password
+    |> hash_password()
+    |> occurrences_of_hash()
+    |> Kernel.>=(@weakness_threshold)
+  end
+
+  defp hash_password(string_password) do
+    :sha
+    |> :crypto.hash(string_password)
+    |> Base.encode16()
+    |> String.upcase()
+  end
+
+  defp range(searchable_range) do
+    url = @base_url <> "range/#{searchable_range}"
+    headers = [{"User-Agent", "hexpm"}]
+
+    case :hackney.get(url, headers, "", connect_timeout: @timeout, recv_timeout: @timeout) do
+      {:ok, 200, _headers, ref} ->
+        {:ok, body} = :hackney.body(ref)
```

Comment:
If we read all of the body unconditionally anyway, I think passing the `with_body` option to hackney call might be simpler.

Reply:
I'll check that out, I wasnt aware of it.
It's going to take me a while to get used to looking at Erlang docs 🤪